pkg/api/types.go

```diff
@@ -1042,6 +1042,10 @@ type AzureFileVolumeSource struct {
 	// the ReadOnly setting in VolumeMounts.
 	// +optional
 	ReadOnly bool
+	// the namespace of the secret that contains Azure Storage Account Name and Key
+	// default is the same as the Pod
+	// +optional
+	SecretNameSpace string
```

Comment:
this is the same discussion we just had with the storageos plugin - see https://github.com/kubernetes/kubernetes/pull/42156#discussion_r120627403